webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/UserSessionInvalidationFilter.scala

```diff
@@ -60,24 +60,24 @@ import zio.syntax.*
  */
 class UserSessionInvalidationFilter(userRepository: UserRepository, userDetailListProvider: FileUserDetailListProvider)
     extends OncePerRequestFilter {
-  private val userStatuses = Ref
-    .make(Map.empty[String, UserStatus])
-    .map(ref => {
-
-      userDetailListProvider.registerCallback(
-        RudderAuthorizationFileReloadCallback(
-          "user-session-invalidation",
-          (users: ValidatedUserList) => {
-            userRepository
-              .getStatuses(users.users.keys.toList)
-              .flatMap(ref.set)
-          }
-        )
+  import UserSessionInvalidationFilter.*
+
+  private val userCache = {
+    val ref: Array[(String, Int)] = new Array(USERS_REF_SIZE) // an array of users, there are not that much users in Rudder
```

Comment:
Actually you should use a Chunk. It's an array in disguise, but the sizing and rising is done for you, and you don't have to use arbitrary limits that will explode one day and we will be very sad, because a new version of rudder will be needed for that

Reply:
And actually : if the array is not of a simple Java type, it won't be memory compact, because Java is a sad place that needs to wait for the Valhalla. So you can use a map, it's basically the same thing, an array of pointers, and it will easier to understand the code (and I don't think we need to go the path of two arrays and use index for lookup before actually measuring the thing as an hot spot)

Reply:
:+1: I changed it back to a `Ref[Map[String, Int]]` which is indeed a lot easier to manipulate !

Reply:
https://github.com/Normation/rudder/pull/5866/commits/22713762017fed7cc9007b19867c39bdbc81ee74